[#9090] Remove deprecated parameters on h2.H2Connection

alexwlchan opened the pull request:
Using explicit `client_side` and `header_encoding` parameters on
`H2Connection` was deprecated in h2 2.5.0, and is being removed
entirely in h2 3.0.  Switch to the new behaviour, which is using
an `H2Configuration` instance.

https://twistedmatrix.com/trac/ticket/9090#ticket

Lukasa commented:
Hey @alexwlchan! As a note for future Twisted contribution you should mark your Trac ticket as "awaiting review" by adding the "review" keyword and reassigning it to no-one (the empty space at the top of the list of names in the dropdown). However, I'll review this myself for now.

Lukasa reviewed (changes requested):
The code changes look fine, however you need to add a topfile which is used to modify the Twisted changelog. In this case, you'll want to put an empty file called `9090.misc` in `src/twisted/web/topfiles`. Mind doing that?

alexwlchan committed:
Remove deprecated parameters on h2.H2Connection

Using explicit `client_side` and `header_encoding` parameters on
`H2Connection` was deprecated in h2 2.5.0, and is being removed
entirely in h2 3.0.  Switch to the new behaviour, which is using
an `H2Configuration` instance.

alexwlchan commented:
> In this case, you'll want to put an empty file called `9090.misc` in `src/twisted/web/topfiles`. Mind doing that?

Done.

Lukasa commented:
Ok, I'm going to push this to the Twisted buildbots so we get a full build.

Lukasa reviewed (approved):
Ok, this looks good to me. Thanks @alexwlchan! :sparkles:
